src/multienc.rs

```diff
@@ -140,15 +146,21 @@ where
 ///
 /// This method expects the ciphertext and the metadata to be concatenated in one slice. Use
 /// [`decrypt_in_place`] if you have two separate slices.
-#[cfg(feature = "std")]
 pub fn decrypt(secret: &Secret, data: &[u8]) -> io::Result<Vec<u8>> {
     let header = Header::deserialize_from(data)?;
     let metadata_len = metadata_size(header.num_recipients);
     let total_len = metadata_len
         .checked_add(header.data_len)
         .ok_or_else(|| io::Error::other("overflow when calculating data size"))?;
     if data.len() < total_len {
-        return Err(io::Error::from(io::ErrorKind::UnexpectedEof));
+        #[cfg(feature = "std")]
+        {
+            return Err(io::Error::from(io::ErrorKind::UnexpectedEof));
+        }
+        #[cfg(not(feature = "std"))]
+        {
+            return Err(io::Error);
```

Comment:
To maintain consistency, I think we should implement the `From` trait on this in our shim so that we push this responsibility to the shim.